src/DynamoCore/Graph/Nodes/PortModel.cs

```diff
@@ -535,6 +535,11 @@ internal string GetOutPortType()
 
             return null;
         }
+
+        internal bool CanAutoCompleteInput()
+        {
+            return !(PortType == PortType.Input && Connectors?.FirstOrDefault()?.Start?.Owner != null);
```

Comment:
Can this be folded in `CanAutoComplete()`? That function is the default precheck in the AutoComplete command

Reply:
Not really, at least not with the current setup. The reason is that AC command is actually chained with NodeConnect command which kicks in at first click ( and worth noting that if CanAutoComplete returns false the node connect command continues ). The NodeConnect command disconnects the port. Later on there is an undo operation in the AutoComplete function that basically connects back the port ( and at this point we can do the check ). 
Not sure why it was done this way - I suspect due to the way Dynamo handles windows messages etc - something to explore if we can break this chaining and actually have NoConnect and AutoComplete command as separate commands.

Reply:
Over all looks good, this chain is still not so clear to me though, would love to talk to you more about it